lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -67,10 +71,64 @@ def destroy
     output = powershell(powershell_args, script_content)
     Puppet.debug "Destroy Dsc Resource returned: #{output}"
     data = JSON.parse(output)
+
     fail(data['errormessage']) if !data['errormessage'].empty?
-    true
+
+    initiate_reboot if data['rebootrequired'] == true
+
+    data
+  end
+
+  def initiate_reboot
```

Comment:
You're not really initiating a reboot here.  You're setting a status on another resource, that will later perform the reboot.